From c17fdad9f13d10b67137d8ae9170d0623b0b3a31 Mon Sep 17 00:00:00 2001
From: Yazan Shhady <yazan.shhady@solid-run.com>
Date: Mon, 7 Aug 2023 20:39:26 +0300
Subject: [PATCH] rzg2l: set memory size and eth pinctl in rzg2l dts

- set memory size for 1GB
- set ethenret pinctl for eth0 & eth1
---
 arch/arm/dts/rzg2l-solidrun.dts | 52 ++++++++++++++++++++++++++++-----
 1 file changed, 44 insertions(+), 8 deletions(-)

diff --git a/arch/arm/dts/rzg2l-solidrun.dts b/arch/arm/dts/rzg2l-solidrun.dts
index 79e9f8522c..47bb4cef77 100644
--- a/arch/arm/dts/rzg2l-solidrun.dts
+++ b/arch/arm/dts/rzg2l-solidrun.dts
@@ -27,33 +27,57 @@
 	memory@48000000 {
 		device_type = "memory";
 		/* first 128MB is reserved for secure area. */
-		reg = <0 0x48000000 0 0x78000000>;
+		reg = <0 0x48000000 0 0x38000000>;
 	};
 
 };
 
 &pinctrl {
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog>;
+	pinctrl_hog: hoggrp {
+		pinmux = <RZG2L_PINMUX(22, 1, 0)>, /* SD0_DEV_SEL_SW */
+				<RZG2L_PINMUX(23, 0, 0)>, /* BT_REG_ON */
+				<RZG2L_PINMUX(23, 1, 0)>; /* WL_REG_ON */ 
+	};
+
 	eth0_pins: eth0 {
 		pinmux = <RZG2L_PINMUX(20, 0, 1)>, //ET0_TXC_TX_CLK
 				<RZG2L_PINMUX(20, 1, 1)>, // ET0_TX_CTL_TX_EN
 				<RZG2L_PINMUX(20, 2, 1)>, // ET0_TXD0
 				<RZG2L_PINMUX(21, 0, 1)>, // ET0_TXD1
 				<RZG2L_PINMUX(21, 1, 1)>, // ET0_TXD2
-				<RZG2L_PINMUX(22, 0, 1)>, // ET0_TXD3
-				<RZG2L_PINMUX(22, 1, 1)>, // ET0_TX_ERR
-				<RZG2L_PINMUX(23, 0, 1)>, // ET0_TX_COL
-				<RZG2L_PINMUX(23, 1, 1)>, // ET0_TX_CRS
+				<RZG2L_PINMUX(22, 0, 1)>, // ET0_TXD3			
 				<RZG2L_PINMUX(24, 0, 1)>, // ET0_RXC_RX_CLK
 				<RZG2L_PINMUX(24, 1, 1)>, // ET0_RX_CTL_RX_DV
 				<RZG2L_PINMUX(25, 0, 1)>, // ET0_RXD 
 				<RZG2L_PINMUX(25, 1, 1)>, // ET0_RXD1
 				<RZG2L_PINMUX(26, 0, 1)>, // ET0_RXD2
-				<RZG2L_PINMUX(26, 1, 1)>, // ET0_RXD3
-				<RZG2L_PINMUX(27, 0, 1)>, // ET0_RX_ERR
+				<RZG2L_PINMUX(26, 1, 1)>, // ET0_RXD3				
 				<RZG2L_PINMUX(27, 1, 1)>, // ET0_MDC
 				<RZG2L_PINMUX(28, 0, 1)>, // ET0_MDIO
 				<RZG2L_PINMUX(28, 1, 1)>; // ET0_LINKSTA
 	};
+
+  eth1_pins: eth1 {
+    pinmux = <RZG2L_PINMUX(37, 2, 1)>, /* ET1_LINKSTA */
+       <RZG2L_PINMUX(37, 0, 1)>, /* ET1_MDC */
+       <RZG2L_PINMUX(37, 1, 1)>, /* ET1_MDIO */
+       <RZG2L_PINMUX(29, 0, 1)>, /* ET1_TXC */
+       <RZG2L_PINMUX(29, 1, 1)>, /* ET1_TX_CTL */
+       <RZG2L_PINMUX(30, 0, 1)>, /* ET1_TXD0 */
+       <RZG2L_PINMUX(30, 1, 1)>, /* ET1_TXD1 */
+       <RZG2L_PINMUX(31, 0, 1)>, /* ET1_TXD2 */
+       <RZG2L_PINMUX(31, 1, 1)>, /* ET1_TXD3 */
+       <RZG2L_PINMUX(33, 1, 1)>, /* ET1_RXC */
+       <RZG2L_PINMUX(34, 0, 1)>, /* ET1_RX_CTL */
+       <RZG2L_PINMUX(34, 1, 1)>, /* ET1_RXD0 */
+       <RZG2L_PINMUX(35, 0, 1)>, /* ET1_RXD1 */
+       <RZG2L_PINMUX(35, 1, 1)>, /* ET1_RXD2 */
+       <RZG2L_PINMUX(36, 0, 1)>; /* ET1_RXD3 */
+  };
+
 };
 
 &xinclk {
@@ -84,7 +108,7 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&eth0_pins>;
 	phy-handle = <&phy0>;
-	phy-mode = "rgmii";
+	phy-mode = "rgmii-id";
 	status = "okay";
 
 	phy0: ethernet-phy@0 {
@@ -92,6 +116,18 @@
 	};
 };
 
+&eth1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&eth1_pins>;
+	phy-handle = <&phy0>;
+	phy-mode = "rgmii-id";
+	status = "okay";
+
+	phy1: ethernet-phy@0 {
+		reg = <0>;
+	};
+};
+
 &i2c0 {
 	pinctrl-names = "default";
 	status = "okay";
-- 
2.25.1

